### ISSUE TITLE
`suppress_warnings = ['misc.highlighting_failure']` is broken

### ISSUE THREAD

**Issue by quasipedia** [sphinx-doc/sphinx#3336]
Subject: `suppress_warnings = ['misc.highlighting_failure'] ` is broken

### Problem
I'm documenting a react project, which means most of the snippets use JSX, the xml-like extension to javascript.  Albeit pygments support ES6 (the version of JS the project use, and that in turns support JSX), *the JSX syntax itself is not supported by Sphinx*.

Since my CI has doucementation buildind as a step, I produce the docs with the `-W` flag (treat warnings as errors), and I thought to use `suppress_warnings = ['misc.highlighting_failure']` introduced in Sphinx 1.5.0 to get rid of the messages relating to JSX.

However it does not work.  *Despite having set the option as above, I still get the warnings, even if I produce the docs _without_ the `-W` flag*.

#### Procedure to reproduce the problem

in `conf.py`
```
suppress_warnings = ['misc.highlighting_failure'] 
```
in some rst file
```
.. code-block:: javascript

    const Foo = () => (
      <div>
        bar
      </div>
    )
    export default Foo
```

#### Error logs / results
```
PYGMENTS_NODE_COMMAND=node sphinx-build -b html -n -W -d _build/doctrees   doc _build/html
Running Sphinx v1.5.1
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 3 source files that are out of date
updating environment: 3 added, 0 changed, 0 removed
reading sources... [100%] index                                                                                                                                                                                                                                                           
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 33%] coding-style                                                                                                                                                                                                                                                     
Warning, treated as error:
/home/myself/dev/foobar/doc/coding-style.rst:61: WARNING: Could not lex literal_block as "javascript". Highlighting skipped.

Makefile:55: recipe for target 'html' failed
make: *** [html] Error 1
```

#### Expected results
It should create the docs without a pip (pun intended!) instead.

### Environment info
- OS: Fedora 25 64bit
- Python version: 3.5.2
- Sphinx version: 1.5.1
- Zsh version: 5.2  (but the problem persist switching to bash, so I don't think it's relevant)

**Comment by quasipedia** [comment]
Also, for extra points I tried this after performing:

  - `find -name '__pycache__' | xargs rm -rf` in my virtualenv installation
  - `make clean` in my doc directory

**Comment by tk0miya** [comment]
Sorry for inconvenience. It was my typo...
It was fixed at 07633a5a712a3b06c21ef35c341a606ad9b96d47 ant will be released in next release (maybe in a few days).

**Comment by quasipedia** [comment]
Thank you for the swift fix @tk0miya! Looking forward to 1.5.2 then! :)
